versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java

```diff
@@ -150,12 +156,17 @@ void transplant(
    * @param fromHash The hash we are using to get additional commits
    * @param toBranch The branch that we are merging into
    * @param expectedHash The current head of the branch to validate before updating (optional).
+   * @param updateCommitMetadata Function to allow resetting properties application due to merge.
    * @throws ReferenceConflictException if {@code expectedBranchHash} doesn't match the stored hash
    *     for {@code toBranch}
    * @throws ReferenceNotFoundException if {@code toBranch} or {@code fromHash} is not present in
    *     the store.
    */
-  void merge(Hash fromHash, BranchName toBranch, Optional<Hash> expectedHash)
+  void merge(
+      Hash fromHash,
+      BranchName toBranch,
+      Optional<Hash> expectedHash,
+      BiFunction<Serializer<METADATA>, ByteString, ByteString> updateCommitMetadata)
```

Comment:
This should be a `Function<METADATA, METADATA>`

Reply:
The serialiser is available with the StoreWorker, and appears to be available at an implementation layer of the version store `storeWorker.getMetadataSerializer()`. However, the logic to update CommitMeta has to be at TreeResource. 
Hence, expecting a serialiser as input, and supplied by the VersionStore implementation.